### PR TITLE
Metada parser example

### PR DESCRIPTION
**Pull request by dmipx**

### CODE REVIEW

File: utilities/examples/md_parser.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    uint32_t    crc;
+};
+const uint8_t md_mipi_rgb_mode_size = sizeof(struct md_mipi_rgb_mode);
+#pragma pack(pop)
+
+/**\brief metadata_mipi_raw - metadata structure
+ *  layout as transmitted and received by backend */
+struct metadata_mipi_depth_raw
+{
+    struct uvc_header_mipi    header;
```
Comment (Nir-Az) [inline utilities/examples/md_parser.c:117]: No need for `#pragma pack(push, 1)` here?

File: utilities/examples/md_parser.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+{
+    char *fname;
+    char *appname = argv[0];
+    char buf[1024];
+    int n;
+    FILE *fp;
+    struct md_mipi_depth_mode *md;
+    struct metadata_mipi_depth_raw *md_raw;
+    if (argc < 2)
+        printf("Missing file name\n");
```
Comment (Nir-Az) [inline utilities/examples/md_parser.c:131]: I suggest
`This application expect an input file name parameter`
`format is: `
`md_parser <file-name>`

File: utilities/examples/md_parser.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    fp = fopen(fname,"r");
+    if (fp)
+        printf("File found!\n");
+    else {
+        printf("File not found!\n");
+        return 1;
+    }
+    n = fread(buf, sizeof(buf), 1, fp);
+    
+    md_raw = (struct metadata_mipi_depth_raw *)(buf);
```
Comment (Nir-Az) [inline utilities/examples/md_parser.c:143]: This prints a single MD ?

File: utilities/examples/md_parser.c [hunk 4]
```diff
@@ -0,0 +1,177 @@
+/* Metadata parser example for D457.
```
Comment (Nir-Az) [inline utilities/examples/md_parser.c:1]: Make we should place it in 
`utilities/md_parser`?
Your call

File: utilities/examples/md_parser.c (hunk 1 above)
Comment (dmipx) [reply utilities/examples/md_parser.c:117]: no need in specific here, but we will add it.

File: utilities/examples/md_parser.c (hunk 2 above)
Comment (dmipx) [reply utilities/examples/md_parser.c:131]: ok done

File: utilities/examples/md_parser.c (hunk 3 above)
Comment (dmipx) [reply utilities/examples/md_parser.c:143]: for now, yes.

File: utilities/examples/md_parser.c (hunk 4 above)
Comment (dmipx) [reply utilities/examples/md_parser.c:1]: it's too small to have own hierarchy, it's better to have set of small and simple examples.

**Review by Nir-Az — APPROVED** [review]

**Pull request merged**